### PR TITLE
feat: initial logger support

### PR DESCRIPTION
**Pull request by 71zenith**
# Pull Request Template

## Type of change

- [ ] Bug fix
- [ ] Feature
- [ ] Documentation update

## Description

*ramble here*

## Checklist

- [ ] any anime playing
- [ ] bumped version
---
- [ ] next, prev and replay work
- [ ] `-c` history and continue work
- [ ] `-d` downloads work
- [ ] `-s` syncplay works
- [ ] `-q` quality works
- [ ] `-v` vlc works
- [ ] `-e` select episode works
- [ ] `-S` select index works
- [ ] `-r` range selection works
- [ ] `--skip` ani-skip works
- [ ] `--skip-title` ani-skip title argument works
- [ ] `--no-detach` no detach works
- [ ] `--dub` and regular (sub) mode both work
- [ ] all providers return links (not necessarily on a single anime, use debug mode to confirm)
---
- [ ] `-h` help info is up to date
- [ ] Readme is up to date
- [ ] Man page is up to date

## Additional Testcases

- The safe bet: One Piece
- Episode 0: Saenai Heroine no Sodatekata ♭
- Unicode: Saenai Heroine no Sodatekata ♭
- Non-whole episodes: Tensei shitara slime datta ken (ep. 24.5, ep. 24.9)

### CODE REVIEW

**Comment by 71zenith** [comment]
@port19x it seems that journalctl cleans up the logs every few days? do u still wanna proceed with this feature

**Comment by justchokingaround** [comment]
:/

**Comment by port19x** [comment]
> @port19x it seems that journalctl cleans up the logs every few days? do u still wanna proceed with this feature

That's configured by the distro, on arch I still have logs from march.
I'd say let's proceed

**Comment by 71zenith** [comment]
i'd like someone to test this on mac
@justchokingaround free?

**Comment by port19x** [comment]
I have a m1 macbook air that I bought used like 2 weeks ago.
I can do the testing if I remember

**Comment by 71zenith** [comment]
@port19x did u test?

**Comment by port19x** [comment]
I didn't forget to check, but I forgot this is already done, since it's still a draft PR.
I was confused by your inactivity.
Will test later today

**Comment by port19x** [comment]
Not quite there yet. Ani-cli -l hangs
Screenshot 2024-06-26 at 07 28 55

**Comment by 71zenith** [comment]
can u check the docs to see how to print the logs. its hard for me to rely on stack overflow and random forums

**Comment by port19x** [comment]
Yeah, but might not get to it today

**Comment by port19x** [comment]
I had to take care of an elephant this weekend.
I'll see that I get this done sometime this week.

<details>
<summary>the elephant in question</summary>


</details>

**Comment by CoolnsX** [comment]
> I had to take care of an elephant this weekend. I'll see that I get this done sometime this week.
> 
> the elephant in question

Noice, well it's a great language, and it will be like a stain on ur life 🙃

File: ani-cli [hunk 1]
```diff
@@ -379,6 +383,14 @@ while [ $# -gt 0 ]; do
             : >"$histfile"
             exit 0
             ;;
+        -l | --logview)
+            case "$(uname -s)" in
+                Darwin*) log show --predicate 'eventMessage contains "ani-cli"' --info ;;
```
Comment (port19x) [inline ani-cli:388]:
```suggestion
                Darwin*) log show --predicate 'process == "logger"' ;;
```

**Comment by port19x** [comment]
While macs `logger` does accept a `-t` argument, it's broken.
[source](https://apple.stackexchange.com/questions/366382/macos-unified-logging-how-can-i-filter-messages-based-on-tag-on-the-command-lin)
Since almost nothing uses logger on mac, we can just search for logger

**Comment by port19x** [comment]
Let's merge this after the other two PRs that are currently open

**Comment by port19x** [comment]
Given that #1372 has stalled, I'd like to merge this sometime this week.
Ideally after dealing with the two plattform refinements for steamdeck and mac os

**Comment by port19x** [comment]
Closes #1374

**Review by port19x — APPROVED** [review]
Merge after #1398

**Comment by 71zenith** [comment]
finally

**Comment by port19x** [comment]
Github release coming up a little later, gotta make time to write release notes

**Pull request merged**